### PR TITLE
Confirmation data transfer

### PR DESCRIPTION
**Pull request by arndt-s**
Adds transfer of confirmation data (concretely the `cnf` claim) across trust domains.

### CODE REVIEW

**Comment by arndt-s** [comment]
Succeeds #95

File: draft-ietf-oauth-identity-chaining.md [hunk 1]
```diff
@@ -155,6 +158,15 @@ audience
 
 * If the request itself is not valid or if the given resource or audience are unknown, or are unacceptable based on policy, the authorization server MUST deny the request.
 * The authorization server MAY add, remove or change claims. See [Claims transcription](#claims-transcription).
+* If the request contains a sender-constraint `subject_token` the client MUST provide proof of possession or MUST use client authentication. The authorization server MAY deny a request without proof of possession, subject to policy.
```
[the remark this answers is not in the document]
Comment (arndt-s) [reply draft-ietf-oauth-identity-chaining.md:161]: Yes. I believe this sentence is from the previous proposal. Can probably be omitted as you suggest.

[the remark this answers is not in the document]
Comment (arndt-s) [reply draft-ietf-oauth-identity-chaining.md:161]: However, thinking about it a bit more, some AS may completely block requests without proof of possession, even when client authentication is used.

**Review by kburgin3 — APPROVED** [review]

File: draft-ietf-oauth-identity-chaining.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (PieterKas) [reply draft-ietf-oauth-identity-chaining.md:161]: In that case, perhaps we make client authentication mandatory and optionally allows proof of posession:

"If the request contains a sender-constrained `subject_token` the client MUST use client authentication and MAY provide proof of possession. An authorisation server MAY require both client authentication and proof of possession"

or we can also be explicit that you need to provide proof of posession, and if you can't you must do client authentication.

"If the request contains a sender-constrained `subject_token` the client MUST use client authentication if it is unable to perform proof of posession of the key used to constrain the 'subject open'. An authorisation server MAY require both client authentication and proof of possession"

**Pull request closed without merging**